ballerina/listener_utils.bal

```diff
@@ -285,6 +285,64 @@ isolated function forwardMultipartRequestToExecution(map<Upload|Upload[]> fileIn
     }
 }
 
+isolated function getHttpService(Engine gqlEngine, GraphqlServiceConfig? serviceConfig) returns HttpService {
+    final readonly & ListenerAuthConfig[]? authConfigurations = getListenerAuthConfig(serviceConfig).cloneReadOnly();
+    final ContextInit contextInitFunction = getContextInit(serviceConfig);
+    final CorsConfig corsConfig = getCorsConfig(serviceConfig);
+
+    HttpService httpService = @http:ServiceConfig {
+        cors: corsConfig
+    } isolated service object {
```

Comment:
Can't we add other configs such as auth configs here as well? @ldclakmal ?

Reply:
Yes, we should be able to. That way we can remove the complete logic of auth handler creation from GraphQL. But, the only concern I have with adding auth configs here is that the response message in case of auth failure. Because HTTP responds as 401/403 responses, and that is not applicable in GraphQL context. We may have to add interceptor to modify that response if we are doing. So, shall we keep that as it is and check for a better approach if we have?

Reply:
+1
Let's go ahead with this approach for now.